src/LibChorus/sync/Synchronizer.cs

```diff
@@ -245,6 +250,7 @@ public void SetIsOneOfDefaultSyncAddresses(RepositoryAddress address, bool enabl
 
 	   private void SendToOthers(HgRepository repo, List<RepositoryAddress> sourcesToTry, Dictionary<RepositoryAddress, bool> connectionAttempt)
 		{
+			using var activity = LibChorusActivitySource.Value.StartActivity();
```

Comment:
What do this `activity` here in `SendToOthers` (and the one in `PullFromOthers`) do if it isn't being used anywhere with `activity?.[...]`? Do you not want any `activity?.SetStatus(ActivityStatusCode.Error,[...]` or `activity?.AddException([...]` within `ThrowIfCancelPending` or `SendToOneOther` (resp., `PullFromOneSource`)?

Reply:
it adds another activity in the trace. So when `SyncNow` is called, it will be the parent of this activity which tracks the start and end time of the activity allowing us to see what's taking time in `SyncNow`, I didn't bother to track cancellation on this activity because it'll be tracked on the parent anyway.